backend/src/apiserver/server/pipeline_server.go

```diff
@@ -164,13 +189,45 @@ func (s *PipelineServer) ListPipelines(ctx context.Context, request *api.ListPip
 		listPipelineRequests.Inc()
 	}
 
+	filterContext, err := ValidateFilter(request.ResourceReferenceKey)
+	if err != nil {
+		return nil, util.Wrap(err, "Validating filter failed.")
+	}
+
+	/*
+		Assume 3 scenarios and ensure backwards compatibility:
+		1. User does not provide resource reference
+		2. User provides resource reference to public namesapce ""
+		3. User provides resource reference to namespace x
+	*/
+	refKey := filterContext.ReferenceKey
+	if refKey == nil {
+		// In single user mode, apply filter with empty namespace for backward compatibile.
```

Comment:
This is for both multi-user and single-user at this moment right? It's for general backwards-compatibility.

Reply:
Right, but you'll actually want to keep this code here going forward. Otherwise you'll run into issues with KFP standalone.